### PR TITLE
Everywhere: Fix many spelling errors

### PR DESCRIPTION
**Pull request by mjz19910**

### CODE REVIEW

File: Kernel/Interrupts/PIC.cpp [hunk 1]
```diff
@@ -161,7 +161,7 @@ void PIC::remap(u8 offset)
     IO::out8(PIC0_CTL, ICW1_INIT | ICW1_ICW4);
     IO::out8(PIC1_CTL, ICW1_INIT | ICW1_ICW4);
 
-    /* ICW2 (upper 5 bits specify ISR indices, lower 3 idunno) */
+    /* ICW2 (upper 5 bits specify ISR indices, lower 3 unspecified) */
```
Comment (creator1creeper1) [inline Kernel/Interrupts/PIC.cpp:164]: That's a weird formulation. How about "`upper 5 bits specify ISR indices, lower 3 don't specify anything`"?

File: Userland/DevTools/HackStudio/Editor.cpp [hunk 2]
```diff
@@ -261,7 +261,7 @@ void Editor::mousemove_event(GUI::MouseEvent& event)
         set_override_cursor(m_hovering_clickable && event.ctrl() ? Gfx::StandardCursor::Hand : Gfx::StandardCursor::IBeam);
 
     for (auto& span : document().spans()) {
-        bool is_clickable = (highlighter->is_navigatable(span.data) || highlighter->is_identifier(span.data));
+        bool is_clickable = (highlighter->is_navigable(span.data) || highlighter->is_identifier(span.data));
```
Comment (creator1creeper1) [inline Userland/DevTools/HackStudio/Editor.cpp:264]: According to wiktionary, "navigatable" is legitimate. I also think that it's easier to understand than "navigable" if you've never come accross the word before.

File: Kernel/Interrupts/PIC.cpp [hunk 3]
```diff
@@ -188,7 +188,7 @@ UNMAP_AFTER_INIT void PIC::initialize()
...
```
Comment (creator1creeper1) [inline Kernel/Interrupts/PIC.cpp:191]: You missed this instance.

File: Userland/DevTools/HackStudio/Editor.cpp [hunk 4]
```diff
@@ -333,7 +333,7 @@ void Editor::mousedown_event(GUI::MouseEvent& event)
 
     if (auto* span = document().span_at(text_position)) {
         if (highlighter->is_navigatable(span->data)) {
-            on_navigatable_link_click(*span);
+            on_navigable_link_click(*span);
```
Comment (creator1creeper1) [inline Userland/DevTools/HackStudio/Editor.cpp:336]: Please keep `navigatable` everywhere

File: Userland/DevTools/HackStudio/Editor.cpp [hunk 5]
```diff
@@ -565,7 +565,7 @@ void Editor::flush_file_content_to_langauge_server()
         document().text());
 }
 
-void Editor::on_navigatable_link_click(const GUI::TextDocumentSpan& span)
+void Editor::on_navigable_link_click(const GUI::TextDocumentSpan& span)
```
Comment (creator1creeper1) [inline Userland/DevTools/HackStudio/Editor.cpp:568]: See above.

File: Userland/DevTools/HackStudio/Editor.h [hunk 6]
```diff
@@ -68,7 +68,7 @@ class Editor final : public GUI::TextEditor {
 
     void show_documentation_tooltip_if_available(const String&, const Gfx::IntPoint& screen_location);
     void navigate_to_include_if_available(String);
-    void on_navigatable_link_click(const GUI::TextDocumentSpan&);
+    void on_navigable_link_click(const GUI::TextDocumentSpan&);
```
Comment (creator1creeper1) [inline Userland/DevTools/HackStudio/Editor.h:71]: See above.

**Review by creator1creeper1 — COMMENTED** [review]
Thank you for considering my review suggestions. You missed a few places where they would have applied as well :^)

File: Kernel/Interrupts/PIC.cpp (hunk 3 above)
Comment (mjz19910) [reply Kernel/Interrupts/PIC.cpp]: found that instance later yesterday

File: Userland/DevTools/HackStudio/Editor.cpp (hunk 5 above)
Comment (mjz19910) [reply Userland/DevTools/HackStudio/Editor.cpp]: forgot about that one...

**Review by linusg — APPROVED** [review]
Thanks!

**Pull request merged**